osu.Game/Screens/OnlinePlay/Components/OnlinePlayBackgroundScreen.cs

```diff
@@ -89,5 +86,47 @@ public override bool OnExiting(ScreenExitEvent e)
             this.MoveToX(0);
             return result;
         }
+
+        [LongRunningLoad]
+        private partial class OnlineBeatmapBackground : Background
+        {
+            private readonly int beatmapId;
+
+            public OnlineBeatmapBackground(int beatmapId)
+            {
+                this.beatmapId = beatmapId;
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(BeatmapLookupCache lookupCache, LargeTextureStore textures, CancellationToken cancellationToken)
+            {
+                try
+                {
+                    APIBeatmap? beatmap = lookupCache.GetBeatmapAsync(beatmapId, cancellationToken).GetResultSafely();
+                    string? coverImage = beatmap?.BeatmapSet?.Covers.Cover;
+
+                    if (coverImage != null)
+                        Sprite.Texture = textures.Get(coverImage);
+                }
+                catch
+                {
+                }
+            }
+        }
+
+        private partial class DefaultBackground : Background
+        {
+            [Resolved]
+            private BeatmapManager beatmapManager { get; set; } = null!;
+
+            [BackgroundDependencyLoader]
+            private void load()
+            {
+                Sprite.Texture = beatmapManager.DefaultBeatmap.GetBackground();
+            }
+
+            public override bool Equals(Background? other)
+                => other is DefaultBackground;
```

Comment:
What's this doing?

- The base implementation of this equality member seems to be doing the exact same thing as this override
- To the best of my knowledge, `IEquatable<Background>` is not really used outside of `BackgroundScreenDefault`

Reply:
I knew this was gonna be commented on. It was added because this used to be a `public` class, but unused in this case - will remove.

Reply:
Yeah sorry if this is seen as me being a stickler but equality members are similar to `.ToString()` overrides in that once they're in nobody wants to ever touch them again lest there be some stupid breakage somewhere resulting from the removal that you can't reasonably check for because how do you even, so I prefer to crack down on these as soon as possible.